### PR TITLE
    Cleanup the use of Double primitive wrapper

### PR DESCRIPTION
**Pull request by BradWalker**
Cleanup the use of Double primitive wrapper. As of Java 9, the VM can do a better job of handling the
        coversion from a Double primitive to a Double Object or vice versa.
    
        This is an extension of the work done in #2498.
    
        No tests have been touched as those expressly need to manage the conversion.
    
        A couple of places I also cleaned up the Float.

### CODE REVIEW

**Comment by BradWalker** [comment]
Hey @mbien & @matthiasblaesing ,

I was having trouble with #5714, so decided to start fressh..

Very sorry to be delayed in responding. Recently I had a death in the family and I'm having "difficulty" re-engaging in things that once provided me with joy. Not to mention, I've noticed that I've started to make "simple" mistakes in both things like this as well as my professional life.

But, I'm trying. 8-)

So in that regard, I think things are fixed. You might need to give this the once over again. And for that, I apologize.

Let's see if we can get this easy task done.

Thanks for your help/understanding.

**Review by matthiasblaesing — COMMENTED** [review]
@BradWalker no need to apologize. Take the time you need, real life should come first.

Thank you for the update. I left two inline comments. The one in `ClassesListController.java` must be fixed, as it breaks the calculation. The other might or might not be fixed.

File: platform/openide.explorer/src/org/openide/explorer/view/NodeTableModel.java [hunk 1]
```diff
@@ -163,9 +164,9 @@ public void setProperties(Property[] props) {
                     Object o = props[i].getValue(ATTR_ORDER_NUMBER);
 
                     if (o instanceof Integer) {
-                        sort.put(new Double(((Integer) o).doubleValue()), Integer.valueOf(ia));
+                        sort.put(((Integer)o).doubleValue(), ia);
```
[the remark this answers is not in the document]
Comment (BradWalker) [reply platform/openide.explorer/src/org/openide/explorer/view/NodeTableModel.java:167]: Hey @matthiasblaesing , thanks for the feedback.. Should be fixed..

**Review by matthiasblaesing — APPROVED** [review]
Looks sane to me. Thank you.

**Review by mbien — APPROVED** [review]
looks good to me. I am very sorry to hear about your loss.

**Pull request merged**